src/terminal/adapter/MacroBuffer.hpp

```diff
@@ -0,0 +1,80 @@
+/*++
+Copyright (c) Microsoft Corporation
+Licensed under the MIT license.
+
+Module Name:
+- MacrosBuffer.hpp
+
+Abstract:
+- This manages the parsing and storage of macros defined by the DECDMAC control sequence.
+--*/
+
+#pragma once
+
+#include "DispatchTypes.hpp"
+#include <array>
+#include <bitset>
+#include <string>
+
+// fwdecl unittest classes
+#ifdef UNIT_TESTING
+class AdapterTest;
+#endif
+
+namespace Microsoft::Console::VirtualTerminal
+{
+    class StateMachine;
+
+    class MacroBuffer
+    {
+    public:
+        // The original DEC terminals only supported 6K of memory, which is
+        // probably a bit low for modern usage. But we also don't want to make
+        // this value too large, otherwise it could be used in a denial-of-
+        // service attack. So for now this is probably a sufficient limit, but
+        // we may need to increase it in the future if we intend to support
+        // macros containing sixel sequences.
+        static constexpr size_t MAX_SPACE = 0x40000;
+
+        MacroBuffer() = default;
+        ~MacroBuffer() = default;
+
+        size_t GetSpaceAvailable() const noexcept;
+        uint16_t CalculateChecksum() const noexcept;
+        std::wstring_view GetMacroSequence(const size_t macroId) const noexcept;

```

Comment:
Given that `macroId` comes from the VT code, should it use `VTInt` instead?

Reply:
I'm fairly certain I had it like that initially and it ended up failing an audit check because of a signed/unsigned comparison (or something of that sort).